src/quantuminspire/api.py

```diff
@@ -813,7 +813,8 @@ def _generate_error_result(message: str) -> Dict[str, Any]:
     def execute_qasm(self, qasm: str, backend_type: Optional[Union[Dict[str, Any], int, str]] = None,
                      number_of_shots: Optional[int] = None, collect_tries: Optional[int] = None,
                      default_number_of_shots: Optional[int] = None, identifier: Optional[str] = None,
-                     full_state_projection: bool = False) -> Dict[str, Any]:
+                     full_state_projection: bool = False,
+                     user_data : str = '') -> Dict[str, Any]:
```

Comment:
user_data should be added to the Args block. You can copy the text from execute_qasm_async method

Reply:
The `user_data` fix perhaps should have gone into a different PR. I updated the docstring